[Python] Fix five classes of PEP-8 violations (E101/E111/E128/E302/W191)

practicalswift opened the pull request:
<!-- Please complete this template before creating pull request. -->
#### What's in this pull request?
<!-- Description about pull request. -->

This commit increases the consistency for the Python code in the repo by fixing all instances of the following five classes of [PEP-8](https://www.python.org/dev/peps/pep-0008/) violations:
* E101: indentation contains mixed spaces and tabs
* E111: indentation is not a multiple of four
* E128: continuation line under-indented for visual indent
* E302: expected 2 blank lines, found 1
* W191: indentation contains tabs

The corresponding ignore rules are also removed from `.pep8`. This will allow for easy identification of potential future regressions with regards to these violations.

The Python code in the repo is getting very close to being fully in line with idiomatic Python style as recommended by the Python Software Foundation's [Style Guide for Python Code (PEP-8)](https://www.python.org/dev/peps/pep-0008/) :-)

Only two violation classes remain after this commit:
* E402: module level import not at top of file (a total of 19 violations in 7 files)
* E501: line too long (… > 79 characters) (a total of 358 violations in 47 files)

#### Resolved bug number: –
<!-- If this pull request resolves any bugs from Swift bug tracker -->

* * * *

<!-- This selection should only be completed by Swift admin -->
Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

        Platform     | Comment
        ------------ | -------------
        All supported platforms     | @swift-ci Please smoke test
        OS X platform               | @swift-ci Please smoke test OS X platform
        Linux platform              | @swift-ci Please smoke test Linux platform

 **Validation Testing**

        Platform     | Comment
        ------------ | -------------
        All supported platforms     | @swift-ci Please test
        OS X platform               | @swift-ci Please test OS X platform
        Linux platform              | @swift-ci Please test Linux platform

Note: Only members of the Apple organization can trigger swift-ci.
</details>
<!-- Thank you for your contribution to Swift! -->

practicalswift commented:
@modocache Pythonista needed - do you have time to review? That would be greatly appreciated :-)

modocache commented on utils/build-script:
```diff
@@ -306,311 +306,311 @@ details of the setups of other systems or automated environments.""")
     parser.add_argument("--watchos",
-        help="""
+                        help="""
 also build for Apple watchos, but disallow tests that require an watchOS device""",
```
Unrelated, but we should fix the fact that "watchOS" is spelled two different ways in this single sentence :joy:

modocache commented:
I'm a fan of these changes. :smiley: 

Unrelated, but it would be great if we could make it *extremely easy* for contributors to the Python in this repository to conform to these style rules. Ideally, I think that would include:

- Documentation explaining how to validate Python code before committing or sending a pull request. Perhaps an additional checkbox in "before merging this pull request" section of the pull request template?
- Perhaps a `validate.py` program in this repository, so contributors could simply run the program to see if the changes they've made are up to spec?
- A CI step that checks for conformance. Would it be controversial to suggest that CI should fail for pull requests that introduce Python style regressions, @shahmishal? I don't want CI failures to become too commonplace, but it would be nice to keep the Python code in the repo conformant to our style guide, without relying on @practicalswift's "gardening" commits.

I'd prefer if someone else on the Swift team merged this pull request--I'm a fan of these style changes, but this is a large change to a bunch of code, and I don't want to overstep my bounds. :bow:

gribozavr commented on benchmark/scripts/Benchmark_Driver:
```diff
@@ -227,24 +236,28 @@ def submit(args):
-            swift_repo=args.swift_repo)
+                   num_samples=args.iterations, verbose=True,
+                   log_directory=args.output_dir,
+                   swift_repo=args.swift_repo)
```
This alignment style is not very compatible with 80 columns or 4-space indentation.  I would prefer we don't adopt it.

modocache replied:
Alternatively, this could be formatted like so:

```py
run_benchmarks(
    file, benchmarks=args.benchmarks,
    num_samples=args.iterations,
    verbose=True,
    log_directory=args.output_dir,
    swift_repo=args.swift_repo)
```

I think the rule simply requires that each line of parameters begin at the same indentation. The "ideal" indentation here is subjective, but I think this indentation rule and the 80 columns rule can co-exist.

gribozavr replied:
That would be a better way to format, indeed.

> I think this indentation rule and the 80 columns rule can co-exist.

I can't agree.  In the ArgumentParser case below, following this formatting rule pushes the code well beyond 80 columns without a reasonable way to re-format.

practicalswift commented:
@gribozavr @modocache Good points. I've now updated this PR to use the formatting suggested by @modocache:

```python
run_benchmarks(
    file, benchmarks=args.benchmarks,
    num_samples=args.iterations,
    verbose=True,
    log_directory=args.output_dir,
    swift_repo=args.swift_repo)
```

I've also fixed the `watchos` typo noticed by @modocache.

Please let me know if you have any suggestions for further improvements :-)

modocache commented:
Thanks, @practicalswift! As for the indentation: I was trying to explain how the rule worked, not necessarily that those lines should be changed. In any case, I'm +1 for this change. :)

tkremenek commented:
@swift-ci please smoke test

practicalswift commented:
@modocache Yes, I know - I applied the rule as described (not only the specific example) :-)

practicalswift commented:
@tkremenek Seems like all tests passed :white_check_mark: 

Let me know if you see the need for further adjustments :-)

practicalswift commented:
@gribozavr + @modocache - do you see the need for any further improvements? :-)

modocache commented:
I'm in favor of this change, but will hold off on merging due to @gribozavr's strong opposition in https://github.com/apple/swift/pull/1526#discussion_r55116233.

practicalswift commented:
@modocache Oh, I thought that had issue been resolved by the adjustments I made to the PR. Do you or @gribozavr see any 80 column issues with the formatting currently in this PR (after the update)? My ambition was to solve all such issues, but I might have failed in doing so - if so let me know where :-)

practicalswift commented:
How this PR evolved:

* Style in initial version of this PR (with 80 char issue pointed out by @gribozavr):

```python
some_very_long_function_name(foo,
                             bar,
                             baz)
```

* Style in current version of this PR (without 80 char issue AFAIK :-)):

```python
some_very_long_function_name(
    foo,
    bar,
    baz)
```

gribozavr commented:
I don't have concerns with the PR after it was adjusted.

gribozavr commented:
@swift-ci Please test

modocache commented:
Thanks as always, @practicalswift! The Python code in this repository owes you a great debt of gratitude! :bow:

practicalswift commented:
@modocache Thank you for these kind words :blush:
